clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -102,12 +137,19 @@ public boolean equals(Object obj) {
             return false;
         if (port != other.port)
             return false;
+        if (rack == null) {
+            if (other.host != null)
```

Comment:
Good catch. I just maintained equals meaning all fields match. I suspect thats the safest path to go right now.